habitat/tasks/rearrange/actions.py

```diff
@@ -0,0 +1,230 @@
+            closest_marker_idx = np.argmin(
+                np.linalg.norm(ee_pos - marker_pos, axis=-1)
+            )
+            closest_marker_pos = marker_pos[closest_marker_idx]
+            to_marker = np.linalg.norm(ee_pos - closest_marker_pos)
+            if to_marker < self.thresh_dist:
+                self._sim.set_snapped_marker(marker_name[closest_marker_idx])
+
+    def step(self, state, should_step=True, **kwargs):
+        return
+
+        # TODO: CANNOT GET CODE INTEGRATED UNTIL ROBO WRAPPER READY
+        # if state == 1:
+        #    self._grasp()
+        #    self._sim.use_robo.close_gripper()
+        # else:
+        #    self._sim.desnap_object()
+        #    self._sim.use_robo.open_gripper()
+        # if should_step:
+        #    return self._sim.step(HabitatSimActions.MAGIC_GRASP)
+
+
+@registry.register_task_action
+class ArmVelAction(SimulatorTaskAction):
+    @property
+    def action_space(self):
+        return spaces.Box(shape=(7,), low=0, high=1, dtype=np.float32)
+
+    def step(self, vel, should_step=True, **kwargs):
+        # clip from -1 to 1
+        vel = np.clip(vel, -1, 1)
+        vel *= self._config.VEL_CTRL_LIM
+        # TODO: THIS IS DIFFERENT FROM MY CODE. I NEED TO ADD TO THE TARGETS NOT
+        # THE ACTUAL JOINT POSITIONS
+        self._sim: RearrangeSim
+        self._sim.robot.arm_motor_pos = vel + self._sim.robot.arm_motor_pos
+        if should_step:
+            return self._sim.step(HabitatSimActions.ARM_VEL)
+        else:
+            return None
+
+
+@registry.register_task_action
+class BaseVelAction(SimulatorTaskAction):
+    def __init__(self, *args, config, sim: RearrangeSim, **kwargs):
+        super().__init__(*args, config=config, sim=sim, **kwargs)
+        self._sim: RearrangeSim = sim
+        self.base_vel_ctrl = habitat_sim.physics.VelocityControl()
+        self.base_vel_ctrl.controlling_lin_vel = True
+        self.base_vel_ctrl.lin_vel_is_local = True
+        self.base_vel_ctrl.controlling_ang_vel = True
+        self.base_vel_ctrl.ang_vel_is_local = True
+
+        self.end_on_stop = self._config.get("END_ON_STOP", False)
+        self.allow_back = self._config.get("ALLOW_BACK", True)
+
+    @property
+    def action_space(self):
+        lim = 20
+        return spaces.Box(shape=(2,), low=-lim, high=lim, dtype=np.float32)
+
+    def _capture_robo_state(self, robot_id, sim):
+        forces = sim.get_articulated_object_forces(robot_id)
+        vel = sim.get_articulated_object_velocities(robot_id)
+        art_pos = sim.get_articulated_object_positions(robot_id)
+        return {
+            "forces": forces,
+            "vel": vel,
+            "pos": art_pos,
+        }
+
+    def _set_robo_state(self, robot_id, sim: RearrangeSim, set_dat):
+        sim.set_articulated_object_forces(robot_id, set_dat["forces"])
+        sim.set_articulated_object_velocities(robot_id, set_dat["vel"])
+        sim.set_articulated_object_positions(robot_id, set_dat["pos"])
```

Comment:
These sim calls are unused? This API was not carried forward past the prototype into Habitat-sim v0.2.0, so this will fail.